### ISSUE TITLE
Issues compiling web3py to .app or .exe 

### ISSUE THREAD

**Issue by araa47** [ethereum/web3.py#948]
* Version: 4.4.1
* Python: 3.5
* OS: osx

### What was wrong?

I am trying to compile a simple web3 python app that signs some transactions using py2app or pyinstall or cx_Freeze, however, the web3 code seems to be causing issues, the exact issue is caused while trying to import web3.
The issue is highlighted here and seems to be caused by the Crypto module: 

https://github.com/pyinstaller/pyinstaller/issues/3619

### How can it be fixed?

By shifting from Crypto module to Cryptodome from pycryptodomex. I am not sure if this is a necessary fix since the issue is only caused when trying to compile python code to an exe for windows or .app for mac. 

### Current Workaround?

In order to work around this, I am trying to build my own code for signing raw transactions. Until now I have got to the step of generating ABI , getting nonce and building "DATA". However, I am not sure how I would go about in signing the transaction with the private key. I have tried to look through the web3py library to try to get the exact function to sign transactions, but haven't had any luck. 

Currently, I have the DATA needed in JSON format for interacting with my smart contract and also the private key of the address and the address itself. How would I go about signing the transaction using python without depending on web3py? I have tried looking around online to find the exact function but haven't had much luck either to reverse engineer the signing so I can code my own python function that takes in DATA field and Private Key as params, and returns the signed transaction, so I can broadcast using eth_sendRawTransaction jsonRPC. 

```
{'value': 0, 'chainId': 4, 'gas': 45000, 'gasPrice': 5000000000, 'nonce': 6, 'to': '0x4D21F3618bC4C50475D4B30993d80CFfe78B2156', 'data': '0x957a0e210000000000000000000000000000000000000000000000000000000000000037'}
```

**Comment by pipermerriam** [comment]
For now I think this issue is going to be out-of-scope for the type of support we can provide.  Formal windows support has been on our roadmap for a while and I expect we'll get to it sometime in the coming months.  I'm going to close this, but feel free to continue the conversation here if you figure anything out.

**Comment by pipermerriam** [comment]
Take a look in the `ethereum/eth-account` repository to find how transactions can be signed locally.

**Comment by araa47** [comment]
@pipermerriam , thanks, will report back with a fix once I figure out how to successfully compile into app or exe.

**Comment by carver** [comment]
The only dependency I can think of for pycryptodome is the keccak256 hash. `coincurve` is used for all the other signing, etc.

You could experiment with other backends in eth-hash, like `pysha3` (although I think that has problems in windows, too). Also, you could look for an implementation of keccak256 that works on windows and add it as a new backend in eth-hash: https://github.com/ethereum/eth-hash/tree/master/eth_hash/backends

**Comment by hoangduyloc** [comment]
@araa47 @pipermerriam @carver 
Hello, I'm trying to make the application from python in MAC, however, It can be successful due to web3 lib, any of you solve this problem? Can you show me some guides, I really appreciate that.
